From: Stig Sandbeck Mathisen <ssm@debian.org>
Bug-Debian: http://bugs.debian.org/671158
Subject: Declare O_CLOEXEC if it is missing

Debian GNU/kFreeBSD "wheezy" is missing O_CLOEXEC.  As a workaround, it can be
defined to 0.
--- gearmand.orig/util/pidfile.cc
+++ gearmand/util/pidfile.cc
@@ -50,6 +50,13 @@
 #include <sys/types.h>
 #include <unistd.h>
 
+/*
+ * Ensure O_CLOEXEC is defined. (See http://bugs.debian.org/671158).
+ */
+#ifndef O_CLOEXEC
+# define O_CLOEXEC 0
+#endif
+
 extern "C" {
 
   char pid_file[1024 * 4]= { 0 };
